### PR TITLE
feat: Add build_type to github_repository (pages)

### LINKED ISSUE #1267: github_repository pages source should allow for github actions source
### Affected Resource(s)
Please list the resources as a list, for example:
- github_repository

### Expected Behavior
I should be allowed to specify the "source" of my pages enabled github_repository resource to be GitHub Actions, much like I can in the GitHub UI.

### Actual Behavior
I am only allowed to specify my pages source as branch/path.

### PR DESCRIPTION
**Pull request by 0x46616c6b**
Resolves #1267

----

### Before the change?

* It is not possible to configure a GitHub Pages with GitHub Actions

### After the change?

* The `build_type` option is added to `pages` property

----

### Pull request checklist
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Added the appropriate label for the given change

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No

## Open Questions

- How can I ensure if the `build_type = "legacy"` that `source` is not optional anymore? 
- This PR is related with #1617, can it be introduced before the resource is split out?

### CODE REVIEW

**Comment by lukas-tide** [comment]
@0x46616c6b to validate the field you could do the checks [here](https://github.com/0x46616c6b/terraform-provider-github/blob/add-build-type-to-repository-pages/github/resource_github_repository.go#L725)

Validation functions are only supported for basic types, so the only option I have found is to do the checks on creation and update and in this case creation reuses update, so only update would work fine I guess. Maybe @kfcampbell could clarify if that is the case and if this is needed for the PR to get merged

File: website/docs/r/repository.html.markdown [hunk 1]
```diff
@@ -124,7 +124,9 @@ initial repository creation and create the target branch inside of the repositor
 
 The `pages` block supports the following:
 
-* `source` - (Required) The source branch and directory for the rendered Pages site. See [GitHub Pages Source](#github-pages-source) below for details.
+* `source` - (Optional) The source branch and directory for the rendered Pages site. See [GitHub Pages Source](#github-pages-source) below for details.
+
+* `build_type` - (Optional) The type of GitHub Pages site to build. Can be `legacy` or `workflow`. If you use `legacy` as build type you need to set the option `source`.
```
Comment (kfcampbell) [inline website/docs/r/repository.html.markdown:129]: What happens if both source and build_type are present, which one takes precedence? What happens if legacy is set for build_type but source isn't set?

**Comment by kfcampbell** [comment]
I'm okay with the validation approach; not running checks on creation/update should result in an error at the API level that will be passed back to the user in any case. 

@0x46616c6b does this result in a breaking change for users? I'm not sure if providing "legacy" by default on all build_types results in any API-level changes or not.

**Comment by 0x46616c6b** [comment]
I understand that upgrading the provider and applying the changes will cause a difference (in state only), but it won't affect the repository configuration. If this change is considered a breaking change (due to the new key changing the state), we should reflect that too.

File: website/docs/r/repository.html.markdown (hunk 1 above)
Comment (0x46616c6b) [reply website/docs/r/repository.html.markdown:129]: **Allowed Options**

```terraform
resource "github_repository" "example" {
  name = "example"

  pages {
    source {
      branch = "main"
      path = "/"
    }
    build_type = "legacy"
  }
}
```

```terraform
resource "github_repository" "example" {
  name = "example"

  pages {
    source {
      branch = "main"
      path = "/"
    }
  }
}
```

```terraform
resource "github_repository" "example" {
  name = "example"

  pages {
    build_type = "workflow"
  }
}
```

**Problematic / wrong Configuration**

```terraform
resource "github_repository" "example" {
  name = "example"

  pages {
    source {
      branch = "main"
      path = "/"
    }
    build_type = "workflow"
  }
}
```

**Review by kfcampbell — APPROVED** [review]
I'm okay with state changing as a non-breaking change as long as the API/GitHub state remains consistent. Thanks!

**Comment by benlangfeld** [comment]
This appears to have broken some use cases: https://github.com/integrations/terraform-provider-github/issues/1724

**Pull request merged**